src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+import java.io.FileNotFoundException;
+
+/**
+ * The main class of CS2103 iP.
+ * It is the main body of this task tracking software
+ * that drives the main logic.
+ */
+public class Duke {
+    private static final String DEFAULT_PATH = "./data/duke.txt";
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+
+    public static void main(String[] args) {
+        Duke duke = new Duke(DEFAULT_PATH);
+        duke.run();
+    }
+
+    /**
+     * Main logic of Duke,
+     * it starts the conversation loop and continuously reads input from
+     * user and execute corresponding functions until exit instructions are given.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```

Comment:
Should this boolean variable be named in a more intuitive manner? Perhaps `hasExited` might be better?